oak_functions/loader/src/server.rs

```diff
@@ -337,28 +344,28 @@ impl WasmState {
     }
 
     // Finds the extension parsing the given i32 handle. Returns appropriate OakStatus, if
-    // the handle is invalid or no extension with this handle can be found in the extension of
-    // WasmState.
+    // no extension with this handle can be found in the extensions of this WasmState.
     fn find_extension(
         &mut self,
-        handle: AbiExtensionHandle,
+        handle: ExtensionHandle,
     ) -> Result<&mut BoxedExtension, OakStatus> {
-        let handle: ExtensionHandle = ExtensionHandle::from_i32(handle).ok_or({
-            self.log_error(&format!("Fail to parse handle {:?}.", handle));
-            OakStatus::ErrInvalidArgs
-        })?;
-
-        // TODO(#2707): Refactor to return OakStatus::ErrInvalidHandle. Add logging.
-        self.extensions_indices
+        let extension = self
+            .extensions_indices
             .iter_mut()
             .find_map(|(_, extension)| {
```

Comment:
What does the `find_map` add here compared to just `find`? It seems to add a layer of Option, which is removed immediately anyways.

Reply:
`find_map` maps the pair of (_, `extension`) to return just the extension. But, good point I refactored using `values_mut` and inlined.